gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestIntegrationSpec.groovy

```diff
@@ -185,58 +185,6 @@ class CreateManifestIntegrationSpec extends GradleIntegrationSpec {
         file('bar-server/product-dependencies.lock').text.contains 'com.palantir.group:foo-service ($projectVersion, 1.x.x)'
     }
 
-    def "createManifest does not force compilation of sibling projects"() {
```

Comment:
@fawind @esword have we really lost this guarantee? IMO it's pretty critical that running `./gradlew --write-locks` remains fast, partly because it gets run a lot locally (e.g. when resolving merge conflicts or cherry picking things) and also gets run a ton on excavator.

Reply:
Agree with Dan. Is there another solution? Maybe using variants?

Reply:
I had suggested using variants and artifactTransformers offline. I think a solution similar to what we do for diagnostics could work here.

The only issue I see is that it is more challenge to get access to a manifest that to a file in the resource directory. I think we could work around this though by producing _both_ an entry in the JARs manifest (for backwards compatability) and a file in the resources directory that contained the Pdeps of a given JAR. We could then cleanly re-use the existing transforms (https://github.com/palantir/sls-packaging/blob/develop/gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/DiagnosticsManifestPlugin.java#L165-L233) and leverage regular gradle cache validation

Reply:
I believe that writing the information to resource files is the way to go (it was the original design for the additional conjure information which is now being written), though that won't necessarily clear all of this up, especially if we are also writing to the meta-inf file (because we still have the existing problem).

However, there may be an optimization (a.k.a. hack) we can appy for in-project dependencies for lock files.  We do not need to actually compute the minimum version for such when writing lock files - we always write the template variable '$projectVersion'.  So if we think of updating locks separately, then we may be able to avoid compiling and writing jars.  I'll ponder a bit more.

Reply:
I like Felipe's idea of also producing a pdeps artifact in addition to writing it in the JAR manifest as this would potentially simplify configuration.

> though that won't necessarily clear all of this up, especially if we are also writing to the meta-inf file (because we still have the existing problem).

Wouldn't project dependencies use the same version of sls-packaging though so we could rely on them using the new code path and producing an artifact containing the pdeps that does not require compiling the source?

> We do not need to actually compute the minimum version for such when writing lock files - we always write the template variable '$projectVersion'

Maybe I misunderstand but I think we need to know the concrete minimum version for writing the "product-dependencies.lock" file ([ref](https://github.com/palantir/sls-packaging/blob/d5b3149fd2867e9a09b50fb689bac0d1de61ac90/gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java#L227))?

Reply:
in-project dependencies are always changed to the template ([ref](https://github.com/palantir/sls-packaging/blob/d5b3149fd2867e9a09b50fb689bac0d1de61ac90/gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyLockFile.java#L77)), unless there is some comparison I am missing?

Reply:
I think we need to revert this. After this PR, excavator checks will not open PRs if the resulting code does not compile.

Here's an example of a dependency upgrade excavator run on our internal server repo that failed because `compileJava` was run as part of `--write-locks`. This run did not open a PR.
https://e.p.b/api/coordinator/checkoutput/c2vrbesi5esaano38i40.stderr